### ISSUE TITLE
Rename pthread device to 'cpu'?

### ISSUE THREAD

**Issue by pjaaskel** [pocl/pocl#1164]
The name 'pthread' can mislead people: They might think that we execute single work-items in threads although pthreads is used only for kernel level and work-group level parallelism. Perhaps we should just rename the driver to 'CPU' and make it the "default CPU driver of PoCL": Whatever threading library is used for multicore/multithreading is not very relevant, as long as we keep the pthread support for maximizing portability. Any thoughts/objections anyone?

**Comment by jansol** [comment]
"cpu" is definitely a lot more intuitive. Currently "basic" sounds more like it would be the "main" driver than "pthread". Should basic perhaps be renamed to something like "cpu-single-thread" while we are at it?

**Comment by pjaaskel** [comment]
Yep, that was the feedback from @franz here as well. The original idea for 'basic' is to have a PoCL driver that supports online compilation via the LLVM chain, but is as simple (and portable) as possible. This _implies_ single core/thread (no WG/task parallelism), but only implies (the main point being the simplicity). So along those lines "cpu-simple", "cpu-basic" or even "minimal-cpu" could be better.

**Comment by jansol** [comment]
I would keep the "cpu-" part at the start, makes it a bit easier to group it together with the threaded cpu driver when they have the same prefix. So "cpu-minimal" would be good, yeah.

**Comment by franz** [comment]
Renamed the user-facing strings to 'cpu' for pthread and 'cpu-minimal' for basic;  the source file renaming can happen after we have merged everything in the backlog.
